MobileBuy/buy/src/main/java/com/shopify/buy/model/CustomerToken.java

```diff
@@ -64,4 +77,26 @@ public Long getCustomerId() {
     public Date getExpiresAt() {
         return expiresAt;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        CustomerToken that = (CustomerToken) o;
+
+        if (!accessToken.equals(that.accessToken)) return false;
+        if (!customerId.equals(that.customerId)) return false;
+        return expiresAt.equals(that.expiresAt);
```

Comment:
We can't assume the customerId and accessToken are not null here as well, I will fix that.
We know they will not be null from server, but we have no way to prevent someone from creating a token with null values.